typescript-react/src/components/Form/Form.tsx

```diff
@@ -24,11 +24,29 @@ import {
   ErrorType,
   FormType,
 } from './interfaces';
+import { deepKeys } from '../../util/FunctionalUtils/FunctionalUtils';
 
 import styles from './Form.module.css';
 
 const genericError = 'Please correct all the required fields.';
 
+const scrollToFirstError = (errors: FormErrors<FormData, ErrorType> | undefined = {}) => {
+  const fieldNames = deepKeys(errors);
+
+  // select library and datepicker don't have a name attribute on input, so the only way to target them is through data-qa-element-id
+  // data-qa-element-id has an unfortunate naming, and changing it now would be a breaking change, would be much prettier if it was data-element-id
+  // adding another data attribute would just duplicate stuff, as it would have the same value as data-qa-element-id
+  for (const fieldName of fieldNames) {
+    const el = document.querySelectorAll(`[name="${fieldName}"], [data-qa-element-id="${fieldName}"] input`)[0] as HTMLElement;
```

Comment:
This data-qa-element-id looks rather wrong
We need to fix this prerequisite instead of including unknown field reference

Reply:
Anyhow, that was the only way to query form fields that are external libs (like selects, etc), that don't have the `[name="${fieldName}"]` defined on their inputs as they have our own external logic.
Only sure thing with them is that they can (and should be) wrapped with our form field decorator that then defines `data-qa-element-id` for them.

Only other option that I could see was creating a duplicate of data-qa-element-id and naming it different :shrug:

Reply:
I'm not gonna merge this as-is since data-qa is not used by revenj or the compiler

Reply:
Fair enough, but I don't think any solution regarding this is going to be acceptable by that criteria :scream: 
I'll try to think of something, but it's looking pretty bleak

Reply:
All in all it seems this is defined in Revenj :(
But its badly named so I'll let it pass :/